zipline/algorithm.py

```diff
@@ -1379,10 +1379,15 @@ def _can_order_asset(self, asset):
                 # If we are after the asset's end date or auto close date, warn
                 # the user that they can't place an order for this asset, and
                 # return None.
-                log.warn("Cannot place order for {0}, as it has de-listed. "
-                         "Any existing positions for this asset will be "
-                         "liquidated on "
-                         "{1}.".format(asset.symbol, asset.auto_close_date))
+                log.warn(
+                    ("Cannot place order for {asset}, either because "
+                     "your data bundle is not up to date, "
```

Comment:
@FreddieV4 we surface this warning on Quantopian to users who aren't going to have any idea what a bundle is (nor can they do anything to update the bundle).  If a user tries to run a backtest whose end date is after the end date of their bundle, IMO the right thing to is to error at that point and tell them to run over a different period or update the bundle.